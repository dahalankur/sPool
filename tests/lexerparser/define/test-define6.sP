# lambda inside a lambda
(int, bool, thread -> (int -> int)) otherFun = lambda (int -> int) (int i, bool b, thread t):
                                                    return lambda int (int j): return i + j
                                                    ;
                                                ;

# lambda inside a lambda inside a lambda
(int, bool, thread -> (int -> (int -> int))) otherFun = lambda (int -> (int -> int)) (int i, bool b, thread t):
                                                        return lambda (int -> int) (int j):
                                                                return lambda int (int k): return i + j + k
                                                                ;
                                                            ;;