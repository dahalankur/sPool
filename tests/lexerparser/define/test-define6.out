PROGRAM {[ 
    DEFINE(false, ARROWTY(INTTY * BOOLTY * THREADTY -> ARROWTY(INTTY -> INTTY)), otherFun, LAMBDA(ARROWTY(INTTY -> INTTY), FORMALS((INTTY, i), (BOOLTY, b), (THREADTY, t)), [ 
        RETURN(LAMBDA(INTTY, FORMALS((INTTY, j)), [ 
            RETURN(BINOP(VAR(i), PLUS, VAR(j)))]))])) 
    DEFINE(false, ARROWTY(INTTY * BOOLTY * THREADTY -> ARROWTY(INTTY -> ARROWTY(INTTY -> INTTY))), otherFun, LAMBDA(ARROWTY(INTTY -> ARROWTY(INTTY -> INTTY)), FORMALS((INTTY, i), (BOOLTY, b), (THREADTY, t)), [ 
        RETURN(LAMBDA(ARROWTY(INTTY -> INTTY), FORMALS((INTTY, j)), [ 
            RETURN(LAMBDA(INTTY, FORMALS((INTTY, k)), [ 
                RETURN(BINOP(BINOP(VAR(i), PLUS, VAR(j)), PLUS, VAR(k)))]))]))]))]}
